### PR TITLE
Bug 1814506 - Add unit tests to cover multi selection of tabs

### PR DESCRIPTION
**Pull request by DreVla**
Three new unit tests have been added to cover cases when multiple tabs are selected or unselected.

### Pull Request checklist

- [X] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [X] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

**Comment by DreVla** [comment]
Hey @MozillaNoah I know you are quite busy with tabs tray refactoring and considering that there the code freeze is soon and this is just a simple addition of unit tests, if you have some time, could you please take a quick look so the PR can be merged? Thank you!

File: app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt [hunk 1]
```diff
@@ -543,6 +544,91 @@ class DefaultTabsTrayControllerTest {
         }
     }
 
+    @Test
+    fun `GIVEN no tabs selected and the user is in multi select mode WHEN the user long taps a tab THEN that tab will become selected`() {
```
Comment (MozillaNoah) [inline app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt:548]:
```suggestion
    fun `GIVEN no tabs selected and the user is not in multi select mode WHEN the user long taps a tab THEN that tab will become selected`() {
```

doesn't this test cover the first scenario of the `when` in `handleMultiSelectClicked` when the user is *not* in multi select mode?

**Review by MozillaNoah — APPROVED** [review]
LGTM! Just one comment about one test's name

**Comment by DreVla** [comment]
Even though this is approved, since it is code freeze day, we will not land it until after the Fenix migration.

**Comment by gabrielluong** [comment]
The monorepo team is working to close out opened PRs before the migration and archiving this repo. Since this is just a test change and to help avoid re-applying the changes on the monorepo, I am gonna push this to landing today.

File: app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt [hunk 2]
```diff
@@ -543,6 +544,91 @@ class DefaultTabsTrayControllerTest {
...
+        controller.handleMultiSelectClicked(tab1, selectionHolder, "Tabs tray")
+        verify(exactly = 1) { controller.handleTabUnselected(tab1) }
+
+        controller.handleMultiSelectClicked(tab2, selectionHolder, "Tabs tray")
+        verify(exactly = 1) { controller.handleTabUnselected(tab2) }
+    }
+
+    @Test
+    fun `GIVEN at least a tab is selected and the user is in multi select mode WHEN the user taps a tab THEN that tab will become selected`() {
+        trayStore = spyk(TabsTrayStore())
```
Comment (jonalmeida) [inline app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt:607]: @DreVla thanks for writing these tests, they're great!

One comment here that I think we should fix (after the monorepo migration is done): if we need to verify an action was sent to a store. We should use the [`CaptureActionsMiddleware`](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/support/test-libstate/src/main/java/mozilla/components/support/test/middleware/CaptureActionsMiddleware.kt) on the store to collect all the actions going the store, then assert with:

```kotlin
middleware.assertLastAction(TabsTrayAction.AddSelectTab::class) {
  assertEquals(tabs, it.tab)
}
```

Here is an [example test class](https://searchfox.org/mozilla-mobile/rev/d2356f2d8b7afc97efcf7db24eb4bc3e0ce246fd/firefox-android/android-components/components/feature/customtabs/src/test/java/mozilla/components/feature/customtabs/CustomTabIntentProcessorTest.kt) that might help as well.

Feel free to use the same bug number as well in the new repo to make the amendment.

cc: @MozillaNoah for reference too.

**Pull request merged**